### ISSUE TITLE
Whitelist for `rxjs-finnish`

### ISSUE THREAD

**Issue by mxchange** [cartant/rxjs-tslint-rules#35]
# Feature request
Allow whitelisting for `rxjs-finnish`, something like:

```json
"rxjs-finnish": [
    true,
    "store"
]
```

So it will ignore those keywords

**Comment by mxchange** [comment]
Perhaps also allowing to pass for which Observable types you want to enable/disable it. And not check methods...

```json
"rxjs-finnish": [
    true,
    { 
      "whitelist": [
         "store"
       ]
     },
     { 
       "exclude": [
         "EventEmitter"
       ]
     },
     { "methods": false }
]
```

Maybe that's asking a bit much, but currently our project only prefers finnish notation on Observable variables, not methods and not EventEmitters...

**Comment by mxchange** [comment]
Also, when you would use this with angular2+ you'd get a lot of warnings for methods like:

- intercept
- resolve

which can't be renamed, because they are part of ng's API

**Comment by cartant** [comment]
I'm not keen on adding configuration for this particular rule. I would prefer to leave it as it is and have exceptions managed using [rule-flag comments](https://palantir.github.io/tslint/usage/rule-flags/):

```ts
@Injectable()
class CanDeactivateSomething implements CanDeactivate<SomeComponent> {
  /*tslint:disable-next-line:rxjs-finnish*/
  canDeactivate(component: SomeComponent, currentRoute: ActivatedRouteSnapshot, currentState: RouterStateSnapshot, nextState: RouterStateSnapshot): Observable<boolean> | Promise<boolean> | boolean {
    ...
  }
}
```

Or:

```ts
@Injectable()
class CanDeactivateSomething implements CanDeactivate<SomeComponent> {
  /*tslint:disable:rxjs-finnish*/
  canDeactivate(
    component: SomeComponent,
    currentRoute: ActivatedRouteSnapshot,
    currentState: RouterStateSnapshot,
    nextState: RouterStateSnapshot
  ): Observable<boolean> | Promise<boolean> | boolean {
    ...
  }
  /*tslint:enable:rxjs-finnish*/
}
```

The rule only applies to the declarations, so there shouldn't be too many flags required. And if there are, perhaps you are better of not using the rule.

**Comment by cartant** [comment]
When I say:

> I'm not keen on adding configuration for this particular rule.

I really do mean that **I'm not keen**. If someone else wants to submit a PR, please do.

**Comment by cartant** [comment]
I've added [options](https://github.com/cartant/rxjs-tslint-rules#rxjs-finnish) for the `rxjs-finnish` rule that will allow enforcement of Finnish notation for combinations of `functions`, `methods`, `parameters`, `properties` and `variables`. I think that this - in combination with rule-flag comments - should be sufficient for most purposes.

**Comment by Jrubzjeknf** [comment]
The rule currently demands that EventEmitters also have the $ postfix. This breaks the default (two-way binding)[https://angular.io/guide/template-syntax#two-way-binding---] of angular. For example, an input variable called `size` requires the output variable to be called `sizeChanged`.

Having to name an EventEmitter to `sizeChanged$` and passing a name to the Output() decorator in order to use two-way binding is just trying to fit the code around the lint rule. Lint rules are supposed to enforce good practice. This is going in the opposite direction.

Would you reconsider making an exception for EventEmitter?

**Comment by cartant** [comment]
@Jrubzjeknf Yes. I think that's entirely reasonable.

The fact that `EventEmitter` is derived from `Subject` is an [implementation detail](https://stackoverflow.com/a/36076701/6680611), so you are right and it should be white listed.

Thanks.

**Comment by Jrubzjeknf** [comment]
Awesome, thank you!

**Comment by cartant** [comment]
@mxchange I've published 4.1.1 which fixes an error with the configuration of the default options for `rxjs-no-finnish` and [adds the interfaces](https://github.com/cartant/rxjs-tslint-rules/blob/v4.1.1/source/rules/rxjsFinnishRule.ts#L12-L13) you mentioned at the beginning of this issue.
